src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -2241,4 +2244,48 @@ public Response getBannerMessages(@PathParam("id") Long id) throws WrappedRespon
                 .collect(toJsonArray()));
 
     }
+    
+    @POST
+    @Consumes("application/json")
+    @Path("/requestSignedUrl")
+    public Response getSignedUrl(JsonObject urlInfo) throws WrappedResponse {
```

Comment:
It is intended for use by other applications but is not used by OpenDP. It offers more flexibility in that a trusted app with an (superuser) API key could request signedUrls as needed rather than having to predefine/register all allowed URLs. Not sure if it will have a use with an SPA or not.

It is now in the guides. As for tests, it doesn't really do anything that isn't tested by the unit tests, so a test could be added, but it would be pretty shallow.

Reply:
Thanks for adding the docs!